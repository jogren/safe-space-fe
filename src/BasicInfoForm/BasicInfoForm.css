.BasicInfoForm {
  width: 100%;
 display: flex;
 justify-content: center;
  align-items: center;
  margin-top: 20px;
  display: inline-block;
}

#fullname-container,
#age-container,
#phone-container {
  width: 50%;
  margin: auto;
  display: flex;
  justify-content: center;
  align-items: center;
}

#fullname, #age, #phone {
  height: 50px;
  border-radius: 5px;
  margin-bottom: 10px;
  margin-left: 20px;
  width: 40%;
  font-size: 20px;
}

#fullname-label, #age-label, #phone-label {
width: 30%;
text-align: end;
font-size: 20px;
}

