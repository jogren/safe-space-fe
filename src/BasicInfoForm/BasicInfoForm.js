import React from 'react'
import './BasicInfoForm.css'

const BasicInfoForm = ({basicInfo, setBasicInfo}) => {
  const handleChange = (e) => setBasicInfo({[e.target.name]: [e.target.value]});

  return (
    <form className="BasicInfoForm">
      <h1 id="basic-info-header">Basic Information:</h1>
      <div id="fullname-container">
        <label id="fullname-label" for="fullname">Full Name:</label>
        <input id="fullname" type="text" placeholder="Add Full Name" onChange={handleChange} name="name" value={basicInfo.name}/>
      </div>
      <div id="age-container">
        <label id="age-label" for="age">Age:</label>
        <input id="age" type='text' placeholder="Add Age" onChange={handleChange} name="age" value={basicInfo.age}/>
      </div>
      <div id="phone-container">
        <label id="phone-label" for="phone">Phone Number:</label>
        <input id="phone" type='text' placeholder="Add Phone Number" onChange={handleChange} name="phone" value={basicInfo.phone}/>
      </div>
    </form>
  )
}

export default BasicInfoForm;
