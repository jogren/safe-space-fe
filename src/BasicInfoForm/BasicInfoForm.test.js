import React from "react";
import { shallow } from "enzyme";
import BasicInfoForm from "./BasicInfoForm";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("BasicInfoForm", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <BasicInfoForm
        personName=""
        setPersonName={jest.fn()}
        personAge=""
        setPersonAge={jest.fn()}
        personPhone=""
        setPersonPhone={jest.fn()}
      />
    );
  });

  it("should render without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match the snapshot with three empty name, age, and phone inputs passed through", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
