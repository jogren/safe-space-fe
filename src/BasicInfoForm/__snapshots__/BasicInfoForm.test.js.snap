// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfoForm should match the snapshot with three empty name, age, and phone inputs passed through 1`] = `
<form
  className="BasicInfoForm"
>
  <h1
    id="basic-info-header"
  >
    Basic Information:
  </h1>
  <div
    id="fullname-container"
  >
    <label
      htmlFor="fullname"
      id="fullname-label"
    >
      Full Name:
    </label>
    <input
      id="fullname"
      name="name"
      onChange={[Function]}
      placeholder="Add Full Name"
      type="text"
      value=""
    />
  </div>
  <div
    id="age-container"
  >
    <label
      htmlFor="age"
      id="age-label"
    >
      Age:
    </label>
    <input
      id="age"
      name="age"
      onChange={[Function]}
      placeholder="Add Age"
      type="text"
      value=""
    />
  </div>
  <div
    id="phone-container"
  >
    <label
      htmlFor="phone"
      id="phone-label"
    >
      Phone Number:
    </label>
    <input
      id="phone"
      name="phone"
      onChange={[Function]}
      placeholder="Add Phone Number"
      type="text"
      value=""
    />
  </div>
</form>
`;
